### ISSUE TITLE
System freeze after running vagrant up

### ISSUE THREAD

**Issue by amitlevy21** [hashicorp/vagrant#11825]
### Vagrant version

```shell
vagrant -v
Vagrant 2.2.9
```

### Host operating system

This is the operating system that you run locally.

Arch Linux
```shell
uname -r
5.8.1-arch1-1
```

### Guest operating system

This is the operating system you run in the virtual machine.
`Arch Linux.`

### Vagrantfile

```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "archlinux/archlinux"

  config.vm.provider "virtualbox" do |vb|
     vb.gui = true
     vb.memory = "512"
     vb.cpus = "1"
  end

end
```

### Debug output

https://gist.github.com/amitlevy21/cfa1a116152b93abceaf8e29fc84e505

### Expected behavior

What should have happened?
`VM should boot normally.`

### Actual behavior

What actually happened?
`System freeze, forced to restart PC.`

### Steps to reproduce

1. Install Arch Linux with 5.8.1 kernel.
2. Install VirutalBox latest and enable it's kernel modules.
```shell
lsmod | grep vbox
vboxnetflt             32768  0
vboxnetadp             28672  0
vboxdrv               528384  2 vboxnetadp,vboxnetflt
```
3. cd to the directory with Vagrantfile and run `vagrant up`
4. The system will freeze after attempting to SSH to the machine and failing to connect.

### Additional notes
1. Iv'e tried changing `vb.memory` and `vb.cpus` to 2048 and 4 respectively. Same result. My PC has 8GB RAM and 4 cores.
2. This is reproducible for me. It happen 4/5 of the times I've tried. Not sure why one time it didn't happen.
3. This is the first time I'm using Vagrant, today I've updated to 5.8.1 Linux kernel, so I don't know if it reproduces in 5.7.x
4. My system does not freeze if I manually run the VM that vagrant built through VirtualBox.

**Comment by RasmusBertell** [comment]
Can confirm this behavior on my machine. Vagrant worked fine until 5.8 kernel. I'm also running Arch Linux with kernel version `5.8.1-arch1-1` and Vagrant version `2.2.9`. Sometimes the system freezes almost immediately after running `vagrant up` and sometimes it freezes during ssh step.

**Comment by soapy1** [comment]
I am unable to reproduce this on the 5.8.3 kernel and virtualbox 6.1.12. Does updating the kernel resolve this issue?

**Comment by amitlevy21** [comment]
The issue persist with kernel 5.8.3 as well and latest virtualbox

**Comment by chrisroberts** [comment]
Hi there,

This is definitely odd behavior, but Vagrant itself should not be able to cause the system to panic. While this behavior does not appear to be isolated (since two of you are here reporting the same behavior) it is not something that is reproducible on any installation. One thing that may provide some useful information is to tail the system logs while you are attempting to bring up the guest with Vagrant to see if anything is reported prior to the system failing.

**Comment by amitlevy21** [comment]
@soapy1 @chrisroberts Thank you for looking into this.

Iv'e used `journalctl -b -1 -x` to extract the logs from yesterday (when I responded that the issue persists with 5.8.3)

https://gist.github.com/amitlevy21/469c5ce60f1d9dda384caaab03d4d886

I ran `vagrant up` at 22:43.

As I understand, the issue does not reproduce with your machines because you are using a different Linux distribution?

**Comment by RasmusBertell** [comment]
I'll try to provide my logs tomorrow, if I remember. I noticed that this issue is only present on my laptop and not desktop, even though both are running Arch and have the same kernel and program versions.

**Comment by goncalor** [comment]
I'm having the same issue described. I am currently running

- Arch Linux
- Linux 5.8.4.arch1-1
- Vagrant 2.2.9-2
- VirtualBox 6.1.12-4

Every time I try to `vagrant up` a box, the host ends up completely freezing after some time. One thing I've managed to reproduce multiple times is the freeze happening when running `ss` on the host when Vagrant is in the SSH phase. From this I suppose the problem network-related.

I have tried this on two different computers and the results are the same (the kernel, etc. are the same though).

> 4. My system does not freeze if I manually run the VM that vagrant built through VirtualBox.

I thought this was the case too, but I've tried to run manually and the freeze still happens, it just seems to take longer (however it's immediate when triggered by `ss`).
